Marlin/src/MarlinCore.cpp

```diff
@@ -888,6 +888,24 @@ void stop() {
  */
 void setup() {
 
+  // TMC Low Power Standby pins need to be set as early as possible or they won't be usable
+  TERN_(HAS_X_TMC_STDBY,SET_INPUT_PULLDOWN(X_STDBY_PIN));
+  TERN_(HAS_X2_TMC_STDBY,SET_INPUT_PULLDOWN(X2_STDBY_PIN));
+  TERN_(HAS_Y_TMC_STDBY,SET_INPUT_PULLDOWN(Y_STDBY_PIN));
+  TERN_(HAS_Y2_TMC_STDBY,SET_INPUT_PULLDOWN(Y2_STDBY_PIN));
+  TERN_(HAS_Z_TMC_STDBY,SET_INPUT_PULLDOWN(Z_STDBY_PIN));
+  TERN_(HAS_Z2_TMC_STDBY,SET_INPUT_PULLDOWN(Z2_STDBY_PIN));
+  TERN_(HAS_Z3_TMC_STDBY,SET_INPUT_PULLDOWN(Z3_STDBY_PIN));
+  TERN_(HAS_Z4_TMC_STDBY,SET_INPUT_PULLDOWN(Z4_STDBY_PIN));
+  TERN_(HAS_E0_TMC_STDBY,SET_INPUT_PULLDOWN(E0_STDBY_PIN));
+  TERN_(HAS_E1_TMC_STDBY,SET_INPUT_PULLDOWN(E1_STDBY_PIN));
+  TERN_(HAS_E2_TMC_STDBY,SET_INPUT_PULLDOWN(E2_STDBY_PIN));
+  TERN_(HAS_E3_TMC_STDBY,SET_INPUT_PULLDOWN(E3_STDBY_PIN));
+  TERN_(HAS_E4_TMC_STDBY,SET_INPUT_PULLDOWN(E4_STDBY_PIN));
+  TERN_(HAS_E5_TMC_STDBY,SET_INPUT_PULLDOWN(E5_STDBY_PIN));
+  TERN_(HAS_E6_TMC_STDBY,SET_INPUT_PULLDOWN(E6_STDBY_PIN));
+  TERN_(HAS_E7_TMC_STDBY,SET_INPUT_PULLDOWN(E7_STDBY_PIN));
+
```

Comment:
These standby pins also have to be setup early in the boot process, so if that block is moved to a more appropriate place, I will need to test it to make sure the TMCs still work.

Reply:
If these are only used when `HAS_TRINAMIC_CONFIG` is set then the init method can be moved to `tmc_util.*`

Reply:
Note also, that Trinamic steppers are only configurable when the stepper drivers are powered by 12-24V. So any initialization that counts on the drivers having power need to be handled in the appropriate place, where other TMC init is done in response to the power being turned on.

Reply:
I wasn't sure where it should go, so I just cleaned it up/expanded what BTT had started with the files they sent me originally.

BigTreeTech will submit a follow-up PR to introduce the low power/standby feature, so I assume that is where this code will eventually go. I don't know when that will be (this PR had to happen ASAP due to issues with Amazon shipping boards early), so this can be moved where ever you think is most appropriate.

Just let me know where & I will throw code on a printer for testing.

Reply:
This is okay to merge now, but it would be good to have a test result before BTT adds on to it. If you can power your board by USB alone then give the standby pins a test while no 12V power is applied. Then retry the test after power is turned on.

Reply:
The TMCs are still happy after moving that standby code when powered from 24V.

> If you can power your board by USB alone then give the standby pins a test while no 12V power is applied. Then retry the test after power is turned on.

I get the normal TMC CONNECTION ERROR when powered by USB only and applying 24v doesn't fix it (though, that doesn't work with the other SKR Mini E3 boards when powered via USB initially either.)

I'm not sure how I'd test the standby pins as they aren't broken out on the board itself.